### PR TITLE
Update get nearby users

### PR DESCRIPTION
**Pull request by mallikaviswas**
* getNearbyUsers now uses GeoFire.
* Filters are also applied. Right now, discoverable and radius are as per the user's settings. Interest filter has been hardcoded as that setting has not yet been included in the Settings Activity. Will do that next.

**Note:** There is a bug when you run the app for the first time for a user. There is some problem with setting user location. The second time you run the app, there are no errors.

### CODE REVIEW

File: app/src/main/java/com/frinder/frinder/activity/DiscoverActivity.java [hunk 1]
```diff
@@ -101,74 +114,109 @@ private void getCurrentLocation() {
...
 
-        //ToDo - transfer this logic to Firebase if possible
+        SharedPreferences pref = PreferenceManager.getDefaultSharedPreferences(this);
+        double searchRadius = pref.getInt("radius", 200);
 
-        ArrayList<User> nearbyUsers = new ArrayList<>();
-        ArrayList<Float> distanceFromCurrentUser = new ArrayList<>();
-        TreeSet<Float> distances = new TreeSet<>();
+        //get filterInterests
+        final ArrayList<String> filterInterests = new ArrayList<>();
```
Comment (donygeorge) [inline app/src/main/java/com/frinder/frinder/activity/DiscoverActivity.java]: This is temporary, right?

File: app/src/main/java/com/frinder/frinder/activity/DiscoverActivity.java [hunk 2]
```diff
@@ -101,74 +114,109 @@ private void getCurrentLocation() {
...
+                                    if (user.getInterests() != null && user.getInterests().size() > 0) {
+                                        ArrayList<String> filterInterestsCopy = new ArrayList<String>(filterInterests);
+                                        filterInterestsCopy.retainAll(user.getInterests());
+                                        if (filterInterestsCopy.size() > 0) interestMatch = true;
+                                        Log.d(TAG, user.getName() + ", common interests =" + filterInterestsCopy.toString());
+                                    }
+                                }
+
+                                //TODO uncomment line below and comment the next one when discoverable field is included
+                                if (isDiscoverable && correctTimestamp && interestMatch) {
```
Comment (donygeorge) [inline app/src/main/java/com/frinder/frinder/activity/DiscoverActivity.java]: Did you mean to comment this line? Ignore if this is what you intended.

Comment (mallikaviswas) [reply app/src/main/java/com/frinder/frinder/activity/DiscoverActivity.java]: I need to remove that TODO comment there. It's already done and works.

File: app/src/main/java/com/frinder/frinder/activity/DiscoverActivity.java (hunk 1 above)
Comment (mallikaviswas) [reply app/src/main/java/com/frinder/frinder/activity/DiscoverActivity.java]: Yes, it is temporary. I will remove it when the filter UI is ready.

**Comment by mallikaviswas** [comment]
I am closing this pull request. The same changes were in a later pull request titled " Auto-refresh of discover users when no one is found in vicinity". That pull request has already been merged into master.

**Pull request closed without merging**